core/test/org/otw/open/controllers/ScreenControllerTest.scala.orig

```diff
@@ -0,0 +1,107 @@
+package org.otw.open.controllers
+
+import com.badlogic.gdx.Screen
+import org.otw.open.{GameScreen, OpenGame}
+import org.otw.open.engine.Engine
+import org.otw.open.engine.impl.{EraserGameEngine, StaticAnimationEngine, CauseAndEffectEngine}
+import org.otw.open.testconfig.UnitSpec
+
+/**
+  * Created by eilievska on 1/28/2016.
+  */
+class ScreenControllerTest extends UnitSpec {
+
+  test("given EraserGameFinished event is dispatched, the returned game screen engine is CauseAndEffectEngine") {
+    val returnedScreenEngine: Engine = ScreenController.dispatchEvent(EraserGameFinished)
+    assert(returnedScreenEngine.isInstanceOf[StaticAnimationEngine])
+  }
+
+  test("given EraserGameFinished event is dispatched,the current game screen engine is CauseAndEffectEngine") {
+    val returnedScreenEngine: Engine = ScreenController.dispatchEvent(EraserGameFinished)
+    val currentGameScreen: GameScreen = getCurrentGameScreen
+    assert(currentGameScreen.engine == returnedScreenEngine)
+  }
+
+  test("given CauseAndEffectFinishedSuccessfully event is dispatched, the returned game screen engine is StaticAnimationEngine") {
+    val returnedScreenEngine: Engine = ScreenController.dispatchEvent(CauseAndEffectFinishedSuccessfully)
+    assert(returnedScreenEngine.isInstanceOf[StaticAnimationEngine])
+  }
+
+  test("given CauseAndEffectFinishedSuccessfully event is dispatched, the current game screen engine is StaticAnimationEngine") {
+    val returnedScreenEngine: Engine = ScreenController.dispatchEvent(CauseAndEffectFinishedSuccessfully)
+    val currentGameScreen: GameScreen = getCurrentGameScreen
+    assert(returnedScreenEngine == currentGameScreen.engine)
+  }
+
+  test("given CauseAndEffectFinishedUnsuccessfully event is dispatched, the returned game screen engine is StaticAnimationEngine") {
+    val returnedScreenEngine: Engine = ScreenController.dispatchEvent(CauseAndEffectFinishedUnsuccessfully)
+    assert(returnedScreenEngine.isInstanceOf[StaticAnimationEngine])
+  }
+
+  test("given CauseAndEffectFinishedUnsuccessfully event is dispatched, the current game screen engine is StaticAnimationEngine") {
+    val returnedScreenEngine: Engine = ScreenController.dispatchEvent(CauseAndEffectFinishedUnsuccessfully)
+    val currentGameScreen: GameScreen = getCurrentGameScreen
+    assert(returnedScreenEngine == currentGameScreen.engine)
+  }
+
+<<<<<<< Updated upstream
```

Comment:
you might want to take another look at the class and remove the extra comments from the git tool you are using